### PR TITLE
Fixup compiler warnings

### PR DESCRIPTION
**Pull request by permcody**
refs #1777

Also, get rid of reinterpret_cast!

### CODE REVIEW

**Comment by permcody** [comment]
I believe that RELAP-7 failures have already been fixed and are not a result of these changes.

File: framework/include/utils/MooseUtils.h [hunk 1]
```diff
@@ -397,6 +397,24 @@ tokenizeAndConvert(const std::string & str,
   return true;
 }
 
+/**
+ * covert takes a string representation of a number type and converts it to the number.
```
Comment (jwpeterson) [inline framework/include/utils/MooseUtils.h]: typo: "covert" -> "convert"

**Review by jwpeterson — CHANGES_REQUESTED** [review]

File: framework/include/utils/MooseUtils.h [hunk 2]
```diff
@@ -397,6 +397,24 @@ tokenizeAndConvert(const std::string & str,
...
+ * This method is here to get around deficiencies in the STL stoi and stod methods where they
+ * might successfully convert part of a string to a number when we'd like to throw and error.
```
Comment (andrsd) [inline framework/include/utils/MooseUtils.h]: did you mean `throw an error`?

**Review by jwpeterson — APPROVED** [review]

**Comment by jwpeterson** [comment]
Only changes were typo fixes, so I'm just gonna merge.

**Pull request merged**